Vagrantfile

```diff
@@ -7,18 +62,76 @@ Vagrant.configure("2") do |config|
     end
     v.memory = 4096
   end
- [
-   %w{centos6.5 chef/centos-6.5},
-   %w{centos7   chef/centos-7.0},
-   %w{ubuntu14  ubuntu/trusty64},
-   %w{ubuntu12  ubuntu/precise64},
-   %w{freebsd10 chef/freebsd-10.0}
- ].each do |machine|
-  (name, box) = machine
+
+  config.vm.provider :aws do |aws, override|
+
+    # Required. Credentials for aws api. vagrant-aws will error if these are unset.
+    aws.access_key_id = ENV['AWS_ACCESS_KEY_ID']
+    aws.secret_access_key = ENV['AWS_SECRET_ACCESS_KEY']
+
+    # Required. Name of aws keypair for launching and accessing the ec2 instance.
+    if [ ENV['AWS_KEYPAIR_NAME'] ]
+      aws.keypair_name = ENV['AWS_KEYPAIR_NAME']
+    end
+
+    # Path to private key for above keypair_name.
+    override.ssh.private_key_path = ENV['AWS_SSH_PRIVATE_KEY_PATH']
+
+    # Name of aws security group that allows tcp/22 from vagrant host.
+    if [ ENV['AWS_SECURITY_GROUP'] ]
+       aws.security_groups = [ ENV['AWS_SECURITY_GROUP'] ]
+    end
+
+    # Set this to the aws region the ec2 instances should be launced in.
+    if ENV['AWS_DEFAULT_REGION']
+      aws.region = ENV['AWS_DEFAULT_REGION']
+    else
+      aws.region = "us-east-1"
+    end
+
+    # Set this to the desired aws instance.
+    if ENV['AWS_INSTANCE_TYPE']
+      aws.instance_type = ENV['AWS_INSTANCE_TYPE']
+    else
+      aws.instance_type = "m3.large"
+    end
+
+    # hack: make aws.region accessible in targets.each do loop.
+    targets["active_region"] = aws.region
+
+  end
+
+targets.each do |name, target|
+  
+  box = target["box"]
+
   config.vm.define name do |build|
 
     build.vm.box = box
 
+    if name.start_with?('aws-')
+      build.vm.provider :aws do |aws, override|
+
+        aws.ami = target['regions'][targets["active_region"]]
+
+        #
+        # user_data is a shell script that is executed by cloud-init early in
+        # the boot process. this script exempts the default ssh user from 
+        # requiretty directive typically set in /etc/sudoers. if this is not 
+        # done, rsync breaks.
+        #
+
+        aws.user_data = "#!/bin/bash\necho 'Defaults:" + target['username'] + " !requiretty' > /etc/sudoers.d/999-vagrant-cloud-init-requiretty && chmod 440 /etc/sudoers.d/999-vagrant-cloud-init-requiretty "
+
+        # OSs have varying default ssh usernames (ec2-user, ubuntu, etc)
+        override.ssh.username = target['username']
```

Comment:
it's my understanding that typical vagrant targets are configured to allow vagrant to login as the user "vagrant". that "vagrant" user is allowed to sudo w/o a password.  these aws targets are not purpose built vagrant targets and do not have a "vagrant" user. they are sock images. however, their default usernames are configured to allow sudo w/o a password. our options are to either 1) build amis that have a "vagrant" user or to 2) tell the vagrant-aws plugin to use a different username to ssh in. 

pros/cons:

* option 1 pros 
 * platform specific usernames in the vagrant file go away
 * the aws.user_data line can go away
* option 1 cons
 * we need to build amis for each os we want an aws-based target for.
 * we need to replicate those ami to all aws regions that we want this to work in.
 * the aws account hosting the amis will incur small ebs fees.
* option 2 pros
 * we dont have to create or replicate amis
 * expanding into other aws regions is easier, just lookup the ami-id and put it in the targets hash
 * no ebs fees
* option 2 cons
 * the aws.user_data work around is required for rsync to work
 * the vagrant file has usernames in it.

there may be other pros/cons/options. what's your take?